DetourCrowd/Source/DetourObstacleAvoidance.cpp

```diff
@@ -331,7 +331,7 @@ float dtObstacleAvoidanceQuery::processSample(const float* vcand, const float cs
 	// find the threshold hit time to bail out based on the early out penalty
 	// (see how the penalty is calculated below to understnad)
 	float minPen = minPenalty - vpen - vcpen;
-	float tThresold = ((double)m_params.weightToi/(double)minPen - 0.1) * (double)m_params.horizTime;
+	float tThresold = (m_params.weightToi / minPen - 0.1f) * m_params.horizTime;
```

Comment:
I am not sure there is any reason to the cast there, but I don't really remember. Rethinking the subject, all the parameter values composing that formula are not very sensitive to mantissa size and have similar exponent:
 - m_params.weightToi, m_params.horizTime: The numeruical values that make sense are in the range of 0.5-5 (2.5 in the demo). 
 - minPen is a sum of the current minPenalty and several normalized difference in velocity (normalized is key here), so it either float max or in the range of 0-20, with large jumps between values due to angular change in sampled velocity.

 

    On Monday, January 11, 2016 8:47 PM, Graham Pentheny <notifications@github.com> wrote:
 

 In DetourCrowd/Source/DetourObstacleAvoidance.cpp:> @@ -331,7 +331,7 @@ float dtObstacleAvoidanceQuery::processSample(const float* vcand, const float cs
>  	// find the threshold hit time to bail out based on the early out penalty
>  	// (see how the penalty is calculated below to understnad)
>  	float minPen = minPenalty - vpen - vcpen;
> -	float tThresold = ((double)m_params.weightToi/(double)minPen - 0.1) * (double)m_params.horizTime;
> +	float tThresold = (m_params.weightToi / minPen - 0.1f) * m_params.horizTime;
Nice article! I definitely learned a bit about FLOPs.It looks like 99d1e79 introduced this. Maybe @axelrodR can help clarify the intent here?—
Reply to this email directly or view it on GitHub.